toolchain/parse/node_kind.def

```diff
@@ -128,7 +128,9 @@ CARBON_PARSE_NODE_KIND_BRACKET(ImportDirective, ImportIntroducer,
 //   _external_: Name or QualifiedDecl
 // Namespace
 CARBON_PARSE_NODE_KIND_CHILD_COUNT(NamespaceStart, 0, CARBON_TOKEN(Namespace))
-CARBON_PARSE_NODE_KIND_CHILD_COUNT(Namespace, 2, CARBON_TOKEN(Semi))
+CARBON_PARSE_NODE_KIND_CHILD_COUNT(Namespace, 2,
+                                   CARBON_TOKEN(Semi)
+                                       CARBON_IF_ERROR(CARBON_TOKEN(Namespace)))
```

Comment:
Can you please add a test of:

```
namespace "x"
```

In this case, won't the location be the string literal token, and still crash? Maybe you should use `CARBON_ANY_TOKEN` here?

Reply:
Good idea, added a bunch of tests.

Interestingly, none of them crash. The parser ends up reusing the namespace token reliably here it seems? Happy to add more tests or make another change though if useful.

Reply:
Thanks, I'm not clear on what the crash was caused by now, maybe I'll look separately.